tests/lib/nested.sh

```diff
@@ -1457,6 +1460,40 @@ nested_start_core_vm_unit() {
                 fi
             fi
         fi
+        nested_setup_vm
+    fi
+}
+
+nested_setup_vm(){
+    if nested_is_core_ge 20; then
+        remote.exec "sudo snap set system journal.persistent=true"
+    fi
+    if [ "${SNAPD_USE_PROXY:-}" = true ]; then
+        nested_no_proxy="${NO_PROXY},10.0.2.2"
+
+        # Add proxy configuration in /etc/environment
+        remote.exec "echo HTTPS_PROXY=$HTTPS_PROXY | sudo tee -a /etc/environment"
+        remote.exec "echo https_proxy=$HTTPS_PROXY | sudo tee -a /etc/environment"
+        remote.exec "echo HTTP_PROXY=$HTTP_PROXY | sudo tee -a /etc/environment"
+        remote.exec "echo http_proxy=$HTTP_PROXY | sudo tee -a /etc/environment"
+        remote.exec "echo NO_PROXY=$nested_no_proxy | sudo tee -a /etc/environment"
+        remote.exec "echo no_proxy=$nested_no_proxy | sudo tee -a /etc/environment"
+        remote.exec "echo SNAPD_USE_PROXY=$SNAPD_USE_PROXY | sudo tee -a /etc/environment"
+
+        # Configure snapd to use the proxy
+        remote.exec "sudo mkdir -p /etc/systemd/system/snapd.service.d"
+        remote.exec "echo [Service] | sudo tee /etc/systemd/system/snapd.service.d/proxy.conf"
+        remote.exec "echo Environment=HTTPS_PROXY=$HTTPS_PROXY HTTP_PROXY=$HTTP_PROXY https_proxy=$HTTPS_PROXY http_proxy=$HTTP_PROXY NO_PROXY=$nested_no_proxy no_proxy=$nested_no_proxy SNAPD_USE_PROXY=$SNAPD_USE_PROXY | sudo tee -a /etc/systemd/system/snapd.service.d/proxy.conf"
+        remote.exec "sudo systemctl daemon-reload"
+        remote.exec "sudo systemctl restart snapd"
+    fi
+    if [ -n "${NTP_SERVER:-}" ]; then
+        # Configure systemd-timesyncd to use the predefined ntp server
+        CONF_FILE="/etc/systemd/timesyncd.conf"
+        remote.exec "sudo sed -i -e '/^NTP=/d' -e '/^FallbackNTP=/d' \"$CONF_FILE\""
+        remote.exec "sudo sed -i '/^\[Time\]/a NTP='\"$NTP_SERVER\" \"$CONF_FILE\""
```

Comment:
Nitpick, but should this be this?

```suggestion
        remote.exec "sudo sed -i '/^\[Time\]/a NTP=\"$NTP_SERVER\"' \"$CONF_FILE\""
```